dcos/dcos-ui
# DCOS-41310, DCOS-41842, DCOS-42506

Update the node count whenever we filter by framework.

Closes https://jira.mesosphere.com/browse/DCOS-41310 .
Closes https://jira.mesosphere.com/browse/DCOS-41842 .
Closes https://jira.mesosphere.com/browse/DCOS-42506 .

## Testing
DCOS-41310:
1. Go to nodes page.
2. Use the "filter by framework" dropdown to filter.
3. Verify that every time it shows the correct number of nodes above the filter (also when using "Clear" to clear).
4. Verify that the added tests make sense.

DCOS-41842:
1. Go to nodes page.
2. Type something in the filter.
3. Click "Clear".
4. Verify that the filter is empty.
5. Select something from the filter dropdown (for example "healthy" or "unhealthy").
6. Click "Clear".
7. Verify that the filter is empty.
8. Verify the same filter is still applied after refreshing the page.
9. Verify that the added test makes sense.

DCOS-42506:
1. Go to nodes page.
2. Switch to graph view.
3. Apply a filter using the dropdown.
4. Switch to grid view.
5. Verify that the filter is kept.
6. Verify that the added test makes sense.

## Trade-offs
None.

## Dependencies
None.

## Screenshots
## https://jira.mesosphere.com/browse/DCOS-41310
### Before
![Peek 2019-06-25 14-01](https://user-images.githubusercontent.com/40791275/60093306-c84ac800-9751-11e9-8888-5041c598c7e9.gif)

### After
![Peek 2019-06-25 14-01-2](https://user-images.githubusercontent.com/40791275/60093314-cbde4f00-9751-11e9-97bf-5212cbab7511.gif)

## https://jira.mesosphere.com/browse/DCOS-41842
### Before
![Peek 2019-06-25 13-53](https://user-images.githubusercontent.com/40791275/60092883-b6b4f080-9750-11e9-8050-507d3ac2132f.gif)

![Peek 2019-06-25 13-50](https://user-images.githubusercontent.com/40791275/60092730-663d9300-9750-11e9-95de-cf2c1551c1ae.gif)

### After
![Peek 2019-06-25 13-53-2](https://user-images.githubusercontent.com/40791275/60092888-bae10e00-9750-11e9-889a-501c54184c0c.gif)

![Peek 2019-07-17 20-14](https://user-images.githubusercontent.com/40791275/61396131-b29f6d00-a8cf-11e9-8928-d806610410b3.gif)

## https://jira.mesosphere.com/browse/DCOS-42506
### Before
![Peek 2019-06-25 13-47](https://user-images.githubusercontent.com/40791275/60092490-d26bc700-974f-11e9-9566-214ed8ad39d8.gif)

### After
![Peek 2019-07-17 20-14-2](https://user-images.githubusercontent.com/40791275/61396135-b7642100-a8cf-11e9-8cae-02fca15f245b.gif)
